addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+    l10n_gr_edi_correlation_id = fields.Many2one(
+        comodel_name='account.move',
+        string='MyDATA Correlated Invoice',
+    )
+    l10n_gr_edi_inv_type = fields.Selection(
+        selection=INVOICE_TYPES_SELECTION,
+        string='MyDATA Invoice Type',
+        compute='_compute_l10n_gr_edi_inv_type',
+        store=True,
+    )
+    l10n_gr_edi_payment_method = fields.Selection(
+        selection=PAYMENT_METHOD_SELECTION,
+        string='MyDATA Payment Method',
+        default='1',
+    )
+    l10n_gr_edi_warnings = fields.Json(compute='_compute_l10n_gr_edi_warnings')
+    l10n_gr_edi_need_correlated = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_need_payment_method = fields.Boolean(compute='_compute_l10n_gr_edi_need_fields')
+    l10n_gr_edi_enable_send_invoices = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_enable_send_expense_classification = fields.Boolean(compute='_compute_l10n_gr_edi_enable_send')
+    l10n_gr_edi_attachment_id = fields.Many2one(comodel_name='ir.attachment')
+
+    ################################################################################
+    # Standard Field Computes
+    ################################################################################
+
+    def unlink(self):
+        self.l10n_gr_edi_document_ids.unlink()
+        return super().unlink()
+
+    @api.depends('l10n_gr_edi_document_ids')
+    def _compute_from_l10n_gr_edi_document_ids(self):
+        self.l10n_gr_edi_state = False
+        self.l10n_gr_edi_mark = False
+        self.l10n_gr_edi_cls_mark = False
+
+        for move in self:
+            if gr_documents := move.l10n_gr_edi_document_ids.sorted():
+                gr_document = gr_documents[0]
+                move.l10n_gr_edi_state = gr_document.state
+                move.l10n_gr_edi_mark = gr_document.mydata_mark
+                move.l10n_gr_edi_cls_mark = gr_document.mydata_cls_mark
+
+                if move.is_purchase_document(include_receipts=True) and not gr_document.mydata_mark:
+                    # For Vendor Bills, look for the first document and get the mark from there
+                    move.l10n_gr_edi_mark = gr_documents.filtered(lambda d: d.state == 'bill_fetched').mydata_mark
+
+    @api.depends('l10n_gr_edi_state')
+    def _compute_show_reset_to_draft_button(self):
+        # EXTENDS 'account'
+        """ Prevent user from resetting the move to draft if it's already sent to MyDATA """
+        super()._compute_show_reset_to_draft_button()
+        for move in self:
+            if move.l10n_gr_edi_state in ('invoice_sent', 'bill_sent'):
+                move.show_reset_to_draft_button = False
+
+    @api.depends('country_code', 'state')
+    def _compute_l10n_gr_edi_warnings(self):
+        for move in self:
+            # Warnings are only calculated when the move state is posted
+            if move.country_code == 'GR' and move.state == 'posted':
```

Comment:
create a method `_l10n_gr_edi_need_mydata` to avoid having those 2 conditions everywhere. I will be a mess if you need to add one. For example, we never check the company currency to be EUR but maybe we should